app/controllers/tasks_controller.rb

```diff
@@ -47,18 +50,25 @@ def update
 
   def destroy
     @task = Task.find(params[:id])
-    if @task.requester === current_user
+    if @task.user == current_user
       @task.destroy
-      redirect_to requesters_tasks_path
+      assign_tasks_and_task_category
       flash[:notice] = "Task Deleted!"
+      render action: '../requesters/tasks/index'
     else
       flash[:notice] = "Sorry, you can't delete this task"
     end
   end
 
   def show
-    @task = Task.find(params[:id])
-    @bid = Bid.new
+    if params[:page] == 'for_requester' && current_user.requester?
+      @task = Task.find(params[:id])
+      @task_cat = @task.task_category
+      render action: '../requesters/tasks/show'
```

Comment:
When this render action was used, we got this error 

```
Missing template tasks/requesters/tasks/index
```
In the views folder tree, we don't have this structure, what we have is  

```
views/requesters/tasks/index

```